discussions-default/src/main/java/org/xwiki/contrib/discussions/internal/DefaultDiscussionService.java

```diff
@@ -77,38 +78,38 @@ public class DefaultDiscussionService implements DiscussionService
     private DiscussionReferencesResolver discussionReferencesResolver;
 
     @Override
-    public Optional<Discussion> create(String applicationHint, String title, String description, String mainDocument,
-        DiscussionStoreConfigurationParameters configurationParameters)
+    public Discussion create(String applicationHint, String title, String description, String mainDocument,
+        DiscussionStoreConfigurationParameters configurationParameters) throws DiscussionException
     {
-        Optional<Discussion> discussion =
-            this.discussionStoreService.create(applicationHint, title, description, mainDocument,
-                configurationParameters).flatMap(this::get);
-        discussion.ifPresent(d -> this.observationManager.notify(new DiscussionEvent(CREATE), applicationHint, d));
+        BaseObject baseObject = this.discussionStoreService.create(applicationHint, title, description, mainDocument,
+            configurationParameters);
+        Discussion discussion = this.mapBaseObject(baseObject);
+        this.observationManager.notify(new DiscussionEvent(CREATE), applicationHint, discussion);
         return discussion;
     }
 
     @Override
-    public Optional<Discussion> getOrCreate(String applicationHint, String title, String description,
+    public Discussion getOrCreate(String applicationHint, String title, String description,
         List<DiscussionContextReference> discussionContexts,
-        DiscussionStoreConfigurationParameters configurationParameters)
+        DiscussionStoreConfigurationParameters configurationParameters) throws DiscussionException
     {
         List<BaseObject> byDiscussionContexts =
             this.discussionStoreService.findByDiscussionContexts(discussionContexts);
+        Discussion result = null;
         if (byDiscussionContexts.isEmpty()) {
-            Optional<DiscussionReference> discussionReferenceOpt =
+            BaseObject baseObject =
```

Comment:
Is the case where baseObject == null handled?

Reply:
baseObject cannot be null: the method either return an instance or throw an exception.